### PR TITLE
Update JSON_INFO.md

### PR DESCRIPTION
**Pull request by RedMisao**
#### Summary
None

#### Purpose of change
Missing documentation for the `cast_spell` use_action

#### Describe the solution
Add the `cast_spell` use_action
Additionally, formatted the `use_action` code block example, a small step towards updating the whole JSON_INFO.md doc:
1. Added pseudo-lint
2. Added indentation to the comments when possible
3. Edited `" :` -> `":`
4. Reduced the material whitelist examples to 5
5. Capitalization, added missing commas, removed final punctuation
6. Moved (optional) to the left and Default to the right when applicable

### CODE REVIEW

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -4367,311 +4367,319 @@ Every item type can have software data, it does not have any behavior:
...
-    "target_charges": 3, // Number of charges the transformed item has.
-    "rand_target_charges": [10, 15, 25], // Randomize the charges the transformed item has. This example has a 50% chance of rng(10, 15) charges and a 50% chance of rng(15, 25). (The endpoints are included.)
-    "ammo_qty": 3,              // If zero or positive set remaining ammo of transformed item to this.
-    "random_ammo_qty": [1, 5],  // If this has values, set remaining ammo of transformed item to one of them chosen at random.
-    "ammo_type": "tape",        // If both this and ammo_qty are specified then set transformed item to this specific ammo.
-    "container": "jar",  // Container holding the target item.
-    "sealed": true,      // Whether the transformed container is sealed; true by default.
-    "menu_text": "Lower visor"      // (Optional) Text displayed in the activation screen, defaults to "Turn on".
-    "moves" : 500         // Moves required to transform the item in excess of a normal action.
+  "type": "transform",                      // The type of method, in this case one that transforms the item
```
Comment (nornagon) [inline doc/JSON_INFO.md:4374]: I think pushing all these comments to the left reduces readability quite substantially.

Comment (RedMisao) [reply doc/JSON_INFO.md:4374]: Sorry, I don't understand. Do you mean to the right?
My goal is to make them look like this:

image

Comment (nornagon) [reply doc/JSON_INFO.md:4374]: This goes to show I shouldn't review code while tired 🤦🏻

File: doc/JSON_INFO.md [hunk 2]
```diff
@@ -4367,311 +4367,319 @@ Every item type can have software data, it does not have any behavior:
...
+  "moves_slow": 1500,     // Number of moves it takes to start a fire on something that is difficult to ignite. This is reduced by survival skill
+  "need_sunlight": true   // Whether the character needs to be in direct sunlight, e.g. to use magnifying glasses
 },
 "use_action": {
-    "type": "unpack", // unpack this item
-    "group": "gobag_contents", // itemgroup this unpacks into
-    "items_fit": true, // Do the armor items in this fit? Defaults to false.
-    "filthy_volume_threshold": "10 L" // If the items unpacked from this item have volume, and this item is filthy, at what amount of held volume should they become filthy
+  "type": "unpack",                 // unpack this item
+  "group": "gobag_contents",        // itemgroup this unpacks into
```
Comment (nornagon) [inline doc/JSON_INFO.md:4515]:
```suggestion
  "type": "unpack",                 // Unpack this item
  "group": "gobag_contents",        // Item group this unpacks into
```

**Pull request merged**